### PR TITLE
Update code to work with Julia 0.7-DEV

### LINKED ISSUE #11: Make compatible with Julia 0.7-DEV
Use [Compat.jl](https://github.com/JuliaLang/Compat.jl) to use the latest Julia syntax and keep it compatible with 0.6.

### PR DESCRIPTION
**Pull request by morris25**
Closes #11

### CODE REVIEW

File: src/lz4frame.jl [hunk 1]
```diff
@@ -257,7 +257,7 @@ or throws an error if it fails.
  """
 function LZ4F_compressUpdate(cctx::Ptr{LZ4F_cctx}, dstBuffer, dstCapacity::Csize_t, srcBuffer, srcSize::Csize_t, cOptPtr)
     check_context_initialized(cctx)
-    ret = ccall((:LZ4F_compressUpdate, liblz4), Csize_t, (Ptr{LZ4F_cctx}, Ptr{Void}, Csize_t, Ptr{Void}, Csize_t, Ptr{LZ4F_compressOptions_t}), cctx, dstBuffer, dstCapacity, srcBuffer, srcSize, cOptPtr)
+    ret = ccall((:LZ4F_compressUpdate, liblz4), Csize_t, (Ptr{LZ4F_cctx}, Ptr{@compat Nothing}, Csize_t, Ptr{@compat Nothing}, Csize_t, Ptr{LZ4F_compressOptions_t}), cctx, dstBuffer, dstCapacity, srcBuffer, srcSize, cOptPtr)
```
Comment (iamed2) [inline src/lz4frame.jl]: Use `Cvoid` instead of `@compat Nothing`

File: src/lz4frame.jl [hunk 2]
```diff
@@ -404,6 +404,6 @@ and start a new one using same context resources.
  """
 function LZ4F_resetDecompressionContext(dctx::Ptr{LZ4F_dctx})
     check_context_initialized(dctx)
-    ccall((:LZ4F_resetDecompressionContext, liblz4), Void, (Ptr{LZ4F_dctx},), dctx)
+    @compat ccall((:LZ4F_resetDecompressionContext, liblz4), Nothing, (Ptr{LZ4F_dctx},), dctx)
```
Comment (iamed2) [inline src/lz4frame.jl]: `Cvoid` works as return type as well

File: test/REQUIRE [hunk 3]
```diff
@@ -0,0 +1 @@
+Compat
```
Comment (iamed2) [inline test/REQUIRE]: You don't need this now that Compat is in the main REQUIRE

File: test/lz4frame.jl [hunk 4]
```diff
@@ -98,7 +99,7 @@
             dctx = Ref{Ptr{CodecLz4.LZ4F_dctx}}(C_NULL)
             srcsize = Ref{Csize_t}(origsize)
             dstsize =  Ref{Csize_t}(8*1280)
-            decbuffer = Vector{UInt8}(1280)
+            decbuffer = @compat Vector{UInt8}(uninitialized, 1280)
```
Comment (iamed2) [inline test/lz4frame.jl]: Why do you used `@compat` here but not in the package code which uses `uninitialized`?

File: test/stream_compression.jl [hunk 5]
```diff
@@ -15,15 +16,15 @@ erat ex bibendum ipsum, sed varius ipsum ipsum vitae dui.
...
+    compressed = transcode(LZ4Compressor, Vector{UInt8}(text))
     @test sizeof(compressed) < sizeof(text)
 
     decompressed = transcode(LZ4Decompressor, compressed)
     @test sizeof(decompressed) > sizeof(compressed)
     @test decompressed == Vector{UInt8}(text)
 
-    test_roundtrip_fileio(LZ4Compressor, LZ4Decompressor)
-    test_roundtrip_transcode(LZ4Compressor, LZ4Decompressor)
+    #test_roundtrip_fileio(LZ4Compressor, LZ4Decompressor)
```
Comment (iamed2) [inline test/stream_compression.jl]: Why commented out?

File: test/stream_compression.jl [hunk 6]
```diff
@@ -15,15 +16,15 @@ erat ex bibendum ipsum, sed varius ipsum ipsum vitae dui.
     @test LZ4CompressorStream <: TranscodingStream
     @test LZ4DecompressorStream <: TranscodingStream
 
-    compressed = transcode(LZ4Compressor, text)
+    compressed = transcode(LZ4Compressor, Vector{UInt8}(text))
```
Comment (iamed2) [inline test/stream_compression.jl]: Make a note of https://github.com/bicycle1885/TranscodingStreams.jl/issues/39 in these so we change this once that issue is closed.

File: test/lz4frame.jl (hunk 4 above)
Comment (morris25) [reply test/lz4frame.jl]: Oddly the ones in the package code didn't seem to be causing any errors when run with 0.6

File: test/stream_compression.jl (hunk 5 above)
Comment (morris25) [reply test/stream_compression.jl]: Because I forgot to uncomment them

File: test/lz4frame.jl (hunk 4 above)
Comment (iamed2) [reply test/lz4frame.jl]: I removed the `@compat` locally and it worked. Can you try removing it and post any errors you get?

Comment (morris25) [reply test/lz4frame.jl]: I remember getting an error earlier, but it seems to have disappeared. I guess I'm just losing my mind.

**Pull request merged**